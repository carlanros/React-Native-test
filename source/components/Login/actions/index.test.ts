import configureMockStore, { MockStore } from 'redux-mock-store';
import { Store } from '../../../store';
import { FacebookLogin } from '../actions';
import * as loginTypes from '../types';

const storeMock = configureMockStore<Store>();

describe('Facebook actions', () => {
    beforeEach(() => {
        // @ts-ignore
        storeMock.clearActions();
     });
      test('FacebookLogin returns Success when done', async () => {
          const expectedReply = {
              name: 'MockName',
              picture: {
                  data: {
                      url: 'http://mockImageUrl/',
                  },
              },
          };
          const expectedPayload = {
            name: expectedReply.name,
            picture: expectedReply.picture.data.url,
            accessToken: 'fbTokenMock',
            type: 'Facebook',
          };
          const expectedActions = [ {type: loginTypes.LoginConstants.LOGIN_SUCCESS,
          payload: expectedPayload,
            },
        ];
        // Dispatch action
        // @ts-ignore
        await storeMock.dispatch(FacebookLogin());

        // @ts-ignore
        expect(storeMock.getActions()).toMatchSnapshot();

        // @ts-ignore
        expect(storeMock.getActions()).toEqual(expectedActions);
      });

});
