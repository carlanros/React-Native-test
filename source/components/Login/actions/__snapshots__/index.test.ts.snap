// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Facebook actions FacebookLogin returns Success when done 1`] = `
Array [
  Object {
    "payload": User {
      "accessToken": "fbTokenMock",
      "name": "test",
      "picture": null,
      "type": "Facebook",
    },
    "type": "LOGIN_SUCCESS",
  },
]
`;
